pkg/installer/packages/manager.go

```diff
@@ -0,0 +1,300 @@
+	err := utils.CheckAvailableDiskSpace(mininumDiskSpace, m.packagesPath)
+	if err != nil {
+		return fmt.Errorf("not enough disk space: %w", err)
+	}
+
+	pkg, err := m.downloader.Download(ctx, url)
+	if err != nil {
+		return fmt.Errorf("could not download package: %w", err)
+	}
+	tmpDir, err := os.MkdirTemp(m.tmpDirPath, fmt.Sprintf("install-experiment-%s-*", pkg.Name)) // * is replaced by a random string
+	if err != nil {
+		return fmt.Errorf("could not create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	configDir := filepath.Join(m.configsDir, pkg.Name)
+	err = extractPackageLayers(pkg.Image, configDir, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not extract package layers: %w", err)
+	}
+	repository := m.repositories.Get(pkg.Name)
+	err = repository.SetExperiment(ctx, pkg.Version, tmpDir)
+	if err != nil {
+		return fmt.Errorf("could not set experiment: %w", err)
+	}
+	return m.startExperiment(ctx, pkg.Name)
+}
+
+// RemoveExperiment removes an experiment.
+func (m *managerImpl) RemoveExperiment(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	repository := m.repositories.Get(pkg)
+	err := repository.DeleteExperiment(ctx)
+	if err != nil {
+		return fmt.Errorf("could not delete experiment: %w", err)
+	}
+	return m.stopExperiment(ctx, pkg)
+}
+
+// PromoteExperiment promotes an experiment to stable.
+func (m *managerImpl) PromoteExperiment(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	repository := m.repositories.Get(pkg)
+	err := repository.PromoteExperiment(ctx)
+	if err != nil {
+		return fmt.Errorf("could not promote experiment: %w", err)
+	}
+	return m.stopExperiment(ctx, pkg)
+}
+
+// Remove uninstalls a package.
+func (m *managerImpl) Remove(ctx context.Context, pkg string) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	err := m.repositories.Delete(ctx, pkg)
+	if err != nil {
+		return fmt.Errorf("could not remove package: %w", err)
+	}
+	return nil
+}
+
+// GarbageCollect removes unused packages.
+func (m *managerImpl) GarbageCollect(ctx context.Context) error {
+	m.m.Lock()
+	defer m.m.Unlock()
+
+	return m.repositories.Cleanup(ctx)
+}
+
+func (m *managerImpl) startExperiment(ctx context.Context, pkg string) error {
+	switch pkg {
+	case packageDatadogAgent:
+		return service.StartAgentExperiment(ctx)
+	case packageDatadogInstaller:
+		return service.StartInstallerExperiment(ctx)
+	default:
+		return nil
+	}
+}
+
+func (m *managerImpl) stopExperiment(ctx context.Context, pkg string) error {
+	switch pkg {
+	case packageDatadogAgent:
+		return service.StopAgentExperiment(ctx)
+	case packageAPMInjector:
+		return service.StopInstallerExperiment(ctx)
+	default:
+		return nil
+	}
+}
+
+func (m *managerImpl) setupUnits(ctx context.Context, pkg string) error {
+	switch pkg {
+	case packageDatadogAgent:
+		return service.SetupAgentUnits(ctx)
+	case packageAPMInjector:
+		return service.SetupAPMInjector(ctx)
+	default:
+		return nil
+	}
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 Suggestion:** ‏Let's the installer OCI setup too (let's do that in another PR)